pymeasure/experiment/results.py

```diff
@@ -129,16 +131,54 @@ def __init__(self, columns, delimiter=','):
         """
         super().__init__()
         self.columns = columns
+        self.units = self._parse_columns(columns)
         self.delimiter = delimiter
 
+    @staticmethod
+    def _parse_columns(columns):
+        """Parse the columns to get units in parenthesis."""
+        units_pattern = r"\((?P<units>[\w/\(\)\*\t]+)\)"
+        units = {}
+        for column in columns:
+            try:
+                match = re.search(units_pattern, column)
+            except TypeError:
+                match = None
+
+            if match:
+                m = match.groupdict()
+                if 'units' in m:
```

Comment:
Can you fail that if check? Is there a way to have a not-None `match`, but then don't match the capture group?